src/gui/Shared/SpectrumAnalyser.cpp

```diff
@@ -41,24 +41,28 @@ void SpectrumAnalyser::visibilityChanged()
 {
     if (isVisible())
     {
-        preTask.setShouldBeRunning (showPreEQ);
-        postTask.setShouldBeRunning (showPostEQ);
+        if (preTask)
+            preTask->get().setShouldBeRunning (showPreEQ);
+        if (postTask)
+            postTask->get().setShouldBeRunning (showPostEQ);
```

Comment:
For this to work correctly, I think we also need some `setter` methods for `showPreEq` and `showPostEQ`, otherwise the task might not get started properly. So we could do something like:
```cpp
void setShouldShowPreEQ (bool shouldShow)
{
    showPreEq = shouldShow;
    if (isVisible() && preTask && ! preTask->get().isTaskRunning())
        preTask->get().setShouldBeRunning (true);
}
```